ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java

```diff
@@ -182,72 +228,63 @@ protected void advanceToken(RemoteReplicaInfo remoteReplicaInfo, ExchangeMetadat
+   * Prints keys absent in server, but present in cloud. This may happen if the blob is compacted or somehow went
+   * missing from the server but is still present in azure backups.
+   * @param rinfo
    */
-  protected EnumSet<MessageInfoType> getBlobStates(MessageInfo messageInfo) {
-    // Blob must be PUT to begin with
-    EnumSet<MessageInfoType> messageInfoTypes = EnumSet.of(MessageInfoType.PUT);
-    if (messageInfo.isDeleted()) {
-      messageInfoTypes.add(MessageInfoType.DELETE);
-    }
-    if (messageInfo.isExpired()) {
-      messageInfoTypes.add(MessageInfoType.EXPIRED);
-    }
-    if (messageInfo.isTtlUpdated()) {
-      messageInfoTypes.add(MessageInfoType.TTL_UPDATE);
-    }
-    if (messageInfo.isUndeleted()) {
-      messageInfoTypes.add(MessageInfoType.UNDELETE);
-    }
-    return messageInfoTypes;
+  public void printKeysAbsentInServer(RemoteReplicaInfo rinfo) {
+    String output = getFilePath(rinfo, BLOB_STATE_MISMATCHES_FILE);
+    azureBlobMap.values().forEach(azureBlob -> {
+      String msg = String.join(BackupCheckerFileManager.COLUMN_SEPARATOR,
+          azureBlob.isEqual(null).stream().map(s -> s.name()).collect(Collectors.joining(",")),
+          MessageInfo.toText(null), MessageInfo.toText(azureBlob),
+          "\n");
+      fileManager.appendToFile(output, msg);
+    });
+    azureBlobMap.clear();
   }
 
   /**
    * Prints a log if local store has caught up with remote store
-   * @param remoteReplicaInfo Info about remote replica
-   * @param exchangeMetadataResponse Metadata response object
+   * @param rinfo Info about remote replica
+   * @param mdResponse Metadata response object
    */
   @Override
-  protected void logReplicationStatus(RemoteReplicaInfo remoteReplicaInfo,
-      ExchangeMetadataResponse exchangeMetadataResponse) {
-    // This will help us know when to stop DR process for sealed partitions
-    fileManager.truncateAndWriteToFile(getFilePath(remoteReplicaInfo, REPLICA_STATUS_FILE), remoteReplicaInfo,
-        exchangeMetadataResponse.localLagFromRemoteInBytes);
+  protected void logReplicationStatus(RemoteReplicaInfo rinfo,
+      ExchangeMetadataResponse mdResponse) {
+    JSONObject json = new JSONObject();
+    try {
+      // Ensures json fields are printed exactly in the order they are inserted
+      Field changeMap = json.getClass().getDeclaredField("map");
+      changeMap.setAccessible(true);
+      changeMap.set(json, new LinkedHashMap<>());
+      changeMap.setAccessible(false);
+    } catch (IllegalAccessException | NoSuchFieldException e) {
+      logger.error(e.getMessage());
+      json = new JSONObject();
+    }
+    // Maintain alphabetical order of fields for string match
+    json.put("replica_token", rinfo.getToken().toString());
+    json.put("replicated_until", rinfo.getReplicatedUntilUTC());
+    // Pretty print with indent for easy viewing
+    fileManager.truncateAndWriteToFile(getFilePath(rinfo, REPLICA_STATUS_FILE), json.toString(4));
```

Comment:
jackson's `ObjectMapper` implementation can respect any order you set for the object. Do you consider using jackson library here. 

1. We have two json libraries, in the code, jackson and org.json, this is annoying.
2. Using a reflection is always something to avoid if we have other ways.

Reply:
i looked at jackson, that is useful if i wanted to print the entire object as json to a file. But here i am only interested in two fields from two different objects, and i don't want to create a third object combining those two just for this purpose.